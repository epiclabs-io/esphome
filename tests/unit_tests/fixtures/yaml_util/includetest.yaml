---
substitutions:
  name: original
  pin: 14

wifi: !include
  file: includes/included.yaml
  vars:
    name: my_custom_ssid

esphome:
  # should be substituted as 'original', not overwritten by vars in the !include above
  name: ${name}
  name_add_mac_suffix: true
  platform: esp8266
  board: !include { file: includes/scalar.yaml, vars: { var1: nodemcu } }

  libraries: !include { file: includes/list.yaml, vars: { var1: Wire } }

#Test jinja works by computing the pin as an expression:
binary_sensor:
  - platform: gpio
    pin: ${pin+1} #should be replaced with '15'
