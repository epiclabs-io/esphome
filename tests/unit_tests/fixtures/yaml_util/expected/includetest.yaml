binary_sensor:
- pin: 15
  platform: gpio
esphome:
  board: nodemcu
  libraries:
  - Wire
  name: original
  name_add_mac_suffix: 1
  platform: esp8266
wifi:
  ssid: my_custom_ssid
